ui/packages/shared/functions/src/useUIFeatureFlag/index.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import useLocalStorageState from 'use-local-storage-state';
+
+const UI_FLAGS = 'ui-flags';
+
+const initializeFlagsFromURL = () => {
+  if (typeof window === 'undefined') {
+    return;
+  }
+
+  const url = new URL(window.location.href);
+  const enableFlag = url.searchParams.get('enable-ui-flag');
+  const disableFlag = url.searchParams.get('disable-ui-flag');
+  if (!enableFlag && !disableFlag) {
+    return;
+  }
+  const flags = JSON.parse(window.localStorage.getItem(UI_FLAGS) ?? '{}');
+  if (enableFlag) {
+    flags[enableFlag] = true;
+  }
+  if (disableFlag) {
+    delete flags[disableFlag];
+  }
+  window.localStorage.setItem(UI_FLAGS, JSON.stringify(flags));
+};
+
+initializeFlagsFromURL();
+
+const useUIFeatureFlag = (featureFlag: string, defaultValue: boolean = false) => {
```

Comment:
could we rename 'featureFlag' to something that indicates that this is the feature name? For example, featureKey or featureName? When I see "featureFlag" I think of the entire name-value pair (the key along with the boolean flag value). Maybe it's just me though 🤔  let me know what you think!